_posts/2015-12-01-pull-requesting-into-pull-requests-for-collaboration.md

```diff
@@ -0,0 +1,58 @@
+
+So, instead of asking someone to introduce further change, it is possible to take a fork of their repository, make the proposed changes yourself, and present it back to them as a pull request into their branch.  If they then choose to accept your pull request, your changes will be added to the original pull request: a real-life example of us doing this workflow is [within the pull request for the "How to DBA" post][pull-request].
+
+### An illustrated example
+
+<div class="grid">
+    <div class="grid-xs-m-1-1 grid-l-1-2 svg-holder">
+        <svg class="git-pr-workflow" version="1.1" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink">
+            <use xlink:href="/images/git-pr-workflow.svg#git-pr-workflow"></use>
+        </svg>
+    </div>
+
+    <div class="grid-xs-m-1-1 grid-l-1-2">
+        <ol class="wrap-break">
+            <li>Alice (up to you whether this is <a href="/authors#alice_kaerast">Alice</a>, or the Alice that is in most hypothetical examples) takes a fork of a source repository's <code>master</code>, and creates a feature branch to work in (<code>git checkout -b feature-branch</code>).</li>
+            <li>Alice does some work in <code>feature-branch</code> and pushes this branch.</li>
+            <li>Alice makes a pull request to have <code>feature-branch</code> reintegrated into the source repository's <code>master</code>.</li>
+            <li>Bob, as reviewer, takes a fork of Alice's repository (<code>git add remote alice git@github.com:alice/skybet.github.io.git</code> followed by <code>git checkout alice/feature-branch</code>)</li>
+            <li>Bob creates some suggested changes to Alice's work and commits to their own feature branch (<code>git checkout -b feature-branch-with-changes</code>) and pushes these changes to their own repository.</li>
+            <li>Bob makes a pull request to Alice to have their <code>feature-branch-with-changes</code> integrated into the original <code>feature-branch</code></li>
+            <li>Alice agrees to the changes and merges the branches.  This creates a new commit in <code>feature-branch</code>.</li>
+            <li>Carol, a new reviewer, sees that there are further changes in the original pull request, and agrees to them.  They merge the original pull request, which brings both Alice's and Bob's work into <code>master</code>.</li>
+        </ol>
+    </div>
+</div>
+
+This isn't a new idea: this isn't revolutionary, but for certain use cases, it works really well - encouraging and inviting collaboration without putting barriers in the way of giving people ownership or freedom over direction.  I'd certainly recommend it if you find yourselves continually batting patch files back and forth between yourselves, making lengthy comments into text boxes, or doing separate tickets for one piece of value so that back-end and front-end development can take place concurrently, for example.
```

Comment:
The last sentence is so long that the trailing "for example" seems out of place.